src/runtime/runtime-debug.cc

```diff
@@ -1014,8 +1014,33 @@ RUNTIME_FUNCTION(Runtime_RecordReplayAssertExecutionProgress) {
     CHECK(gRecordReplayHasCheckpoint);
   }
 
-  recordreplay::Assert("ExecutionProgress %zu %s:%d:%d",
-                       *gProgressCounter, name.c_str(), info.line + 1, info.column);
+  recordreplay::Assert(
+      "JS ExecutionProgress PC=%zu scriptId=%d @%s:%d:%d (%d %d %d %d)",
+      *gProgressCounter, script->id(), name.c_str(), info.line + 1, info.column,
+      recordreplay::AreEventsDisallowed(),
+      !recordreplay::HasDivergedFromRecording(),
+      function->shared().IsUserJavaScript(),
+      function->shared().HasSourceCode());
+
+  if ((RecordReplayIsDivergentUserJSWithoutPause(function->shared())) ||
+      (recordreplay::IsReplaying() && recordreplay::HadMismatch())) {
+    // Print JS stack if user JS was executed non-deterministically
+    // and we were not paused, or if we had a mismatch.
+    if (!gHasPrintedStack) {  // Prevent flood.
+      gHasPrintedStack = true;
+      std::stringstream stack;
+      isolate->PrintCurrentStackTrace(stack);
+
+      recordreplay::Warning(
+          "JS-Stack ExecutionProgress PC=%zu scriptId=%d @%s:%d:%d (%d %d %d "
+          "%d) stack=%s",
```

Comment:
The warning text should have some indication of what problem is being warned about.

Reply:
In both, `Runtime_RecordReplayAssertExecutionProgress` and `Runtime_RecordReplayAssertValue`:

* I added a more descriptive message in case of `RecordReplayIsDivergentUserJSWithoutPause`.
* In case of a mismatch, we already know what is wrong and only use this to provide the additional `JS-Stack` to find out what might have caused the mismatch.
  * Note: If we see this use-case more often, we can consider an `amendment` API that adds information to an Assert, only if it mismatched.